task8-1/AVLTree.c

```diff
@@ -0,0 +1,267 @@
+#include "AVLTree.h"
+#include <stdio.h>
+#include <stdbool.h>
+#include <string.h>
+#include <stdlib.h>
+
+typedef struct Node {
+    char *key;
+    char *value;
+    int balance;
+    struct Node *left;
+    struct Node *right;
+} Node;
+
+typedef struct Tree {
+    Node *root;
+} Tree;
+
+Tree *initTree() {
+    Tree *tree = calloc(1, sizeof(Tree));
+    if (tree == NULL) {
+        printf("Memory allocation error");
+    }
+    return tree;
+}
```

Comment:
```suggestion
Tree *initTree() {
    return calloc(1, sizeof(Tree));
}
```
Никакой печати на консоль в непредназначенных для этого местах. Иначе при переиспользовании этого кода в каком-то большом проекте тут оно будет гадить в консоль непонятными сообщениями, что может расстраивать пользователя, и будет трудно найти, кто именно и зачем гадит.